src/moneypenny/moneypenny.py

```diff
@@ -114,22 +118,30 @@ async def dispatch_order(self, action: str, dossier: Dossier) -> bool:
         container_started : `bool`
             Whether a container needed to be started for this order.  If this
             returns `False`, the action should be considered already
-            complete.
+            complete.  This might be because the order is empty, or it might
+            be because the order has already happened for this dossier and
+            action.
         """
         username = dossier.username
+        cache_key = f"{action}:{dossier}"
+        if self.cache.get(cache_key, None):
+            return False
         self.logger.info(f"Submitting order '{action}' for {username}")
         volumes = self._read_volumes()
         containers = self._read_order(action)
         if not containers:
             self.logger.info("Empty order for {action}, nothing to do")
+            self.cache[cache_key] = True
             return False
+
         await self.k8s_client.make_objects(
             username=username,
             containers=containers,
             volumes=volumes,
             dossier=dossier,
             pull_secret_name=config.docker_secret_name,
         )
+        self.cache[cache_key] = True
```

Comment:
This is premature, isn't it? I think you'd need to wait until the pod actually completed successfully. Otherwise, if the pod fails, you'll still mark this dossier has having run and moneypenny will never try again.